test/e2e/mnist_rayjob_mcad_raycluster_test.go

```diff
@@ -0,0 +1,262 @@
+										Requests: corev1.ResourceList{
+											corev1.ResourceCPU:    resource.MustParse("250m"),
+											corev1.ResourceMemory: resource.MustParse("256Mi"),
+										},
+										Limits: corev1.ResourceList{
+											corev1.ResourceCPU:    resource.MustParse("1"),
+											corev1.ResourceMemory: resource.MustParse("512Mi"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Create an AppWrapper resource
+	aw := &mcadv1beta1.AppWrapper{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ray-cluster",
+			Namespace: namespace.Name,
+		},
+		Spec: mcadv1beta1.AppWrapperSpec{
+			AggrResources: mcadv1beta1.AppWrapperResourceList{
+				GenericItems: []mcadv1beta1.AppWrapperGenericResource{
+					{
+						DesiredAvailable: 1,
+						CustomPodResources: []mcadv1beta1.CustomPodResourceTemplate{
+							{
+								Replicas: 2,
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("250m"),
+									corev1.ResourceMemory: resource.MustParse("512Mi"),
+								},
+								Limits: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("500m"),
+									corev1.ResourceMemory: resource.MustParse("1G"),
+								},
+							},
+						},
+						GenericTemplate: Raw(test, rayCluster),
+					},
+				},
+			},
+		},
+	}
+
+	aw, err = test.Client().MCAD().ArbV1().AppWrappers(namespace.Name).Create(aw)
+	test.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created MCAD %s/%s successfully", aw.Namespace, aw.Name)
+
+	test.T().Logf("Waiting for MCAD %s/%s to be running", aw.Namespace, aw.Name)
+	test.Eventually(AppWrapper(test, namespace, aw.Name), TestTimeoutMedium).
+		Should(WithTransform(AppWrapperState, Equal(mcadv1beta1.AppWrapperStateActive)))
+
+	rayJob := &rayv1alpha1.RayJob{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: rayv1alpha1.GroupVersion.String(),
+			Kind:       "RayJob",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mnist",
+			Namespace: namespace.Name,
+		},
+		Spec: rayv1alpha1.RayJobSpec{
+			Entrypoint: "python /home/ray/jobs/mnist.py",
+			RuntimeEnv: base64.StdEncoding.EncodeToString([]byte(`
+{
+  "pip": [
+    "pytorch_lightning==1.5.10",
+    "torchmetrics==0.9.1",
+    "torchvision==0.12.0"
+  ],
+  "env_vars": {
+  }
+}
+`)),
+			ClusterSelector: map[string]string{
+				RayJobDefaultClusterSelectorKey: rayCluster.Name,
+			},
+			ShutdownAfterJobFinishes: false,
+		},
+	}
+	rayJob, err = test.Client().Ray().RayV1alpha1().RayJobs(namespace.Name).Create(test.Ctx(), rayJob, metav1.CreateOptions{})
+	test.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created RayJob %s/%s successfully", rayJob.Namespace, rayJob.Name)
+
+	test.T().Logf("Waiting for RayJob %s/%s to complete successfully", rayJob.Namespace, rayJob.Name)
+	test.Eventually(RayJob(test, namespace, rayJob.Name), TestTimeoutLong).
+		Should(WithTransform(RayJobStatus, Equal(rayv1alpha1.JobStatusSucceeded)))
```

Comment:
Can you add the check for succeeded or failed job also here?

Reply:
Good catch. Done with e29c28d8e2b36ea8808cca15f242a3ae7741da3b.